### PR TITLE
Resolve some PeerDAS todos

### PR DESCRIPTION
**Pull request by dapplion**
## Issue Addressed

Clean-up some stale `TODO(das)` tags

## Proposed Changes

Explained inlined in code

### CODE REVIEW

File: beacon_node/network/src/sync/manager.rs [hunk 1]
```diff
@@ -1075,22 +1075,14 @@ impl<T: BeaconChainTypes> SyncManager<T> {
...
-        //   accumulate in fork-choice?
-
         match requester {
             SamplingRequester::ImportedBlock(block_root) => {
                 debug!(self.log, "Sampling result"; "block_root" => %block_root, "result" => ?result);
 
-                // TODO(das): Consider moving SamplingResult to the beacon_chain crate and import
-                // here. No need to add too much enum variants, just whatever the beacon_chain or
-                // fork-choice needs to make a decision. Currently the fork-choice only needs to
-                // be notified of successful samplings, i.e. sampling failures don't trigger pruning
```
Comment (dapplion) [inline beacon_node/network/src/sync/manager.rs:1212]: unnecessary comments

File: beacon_node/network/src/sync/network_context.rs [hunk 2]
```diff
@@ -802,7 +801,6 @@ impl<T: BeaconChainTypes> SyncNetworkContext<T> {
                 self.custody_by_root_requests.insert(requester, request);
                 Ok(LookupRequestResult::RequestSent(req_id))
             }
-            // TODO(das): handle this error properly
```
Comment (dapplion) [inline beacon_node/network/src/sync/network_context.rs:777]: This refers to the fact that custody request errors are "nested" but it's okay. So far they are okay to debug

File: beacon_node/network/src/sync/sampling.rs [hunk 3]
```diff
@@ -22,7 +22,6 @@ pub type SamplingResult = Result<(), SamplingError>;
 type DataColumnSidecarList<E> = Vec<Arc<DataColumnSidecar<E>>>;
 
 pub struct Sampling<T: BeaconChainTypes> {
-    // TODO(das): stalled sampling request are never cleaned up
```
Comment (dapplion) [inline beacon_node/network/src/sync/sampling.rs]: Request can't go stale

**Review by jimmygchen — APPROVED** [review]
The old comments clean ups look good to me!

**Comment by jimmygchen** [comment]
@dapplion There are conflicts to fix before merging.

**Comment by jimmygchen** [comment]
@mergify queue

**Pull request merged**